contribute/style-guides/accessibility.md

```diff
@@ -0,0 +1,68 @@
+# Accessibility at Grafana
+
+At Grafana we pay special attention to accessibility and that's why it's important that all components are written with it in
+mind.
+
+[comment]: <> (Add some official standards/checklists we follow.)
+The goal of this document is to list best practices and recommendations when it comes to writing accessible components.
+
+### grafana/ui components
+
+Some grafana/ui components have specific mechanisms built-in that make it easier to write accessible components.
+
+#### Form elements
+
+One of the important accessibility considerations when working with form elements is to make sure form controls are
+properly labelled. For that a `label` element has to be associated with the respective form control. One way to
+do that is to provide `for` attribute to the label that matches the `id` attribute of the form control.
+
+The form components from grafana/ui provide an easier way to achieve that. The form elements, used inside `Field`
+components, will get the `label` properly associated with them given that the element has `id` specified.
+
+As an example, this code
+
+```tsx
+<Field label="Name">
+  <Input id="name" placeholder="Enter a name" />
+</Field>
+```
+
+will be rendered as (simplified)
+
+```html
+<div>
+  <label for="name"> Name </label>
+  <input name="name" type="text" id="name" placeholder="Enter a name" value="" />
+</div>
+```
+
+As long as the form element has a unique `id` attribute specified, it will be automatically accessible when rendered.
+Unfortunately this does not work for all grafana/ui components, most notable exception being `Select`. In those cases adding `aria-label` attribute to the form element is a good option.
```

Comment:
for `Select` we can use `inputId` instead of `id` and the input gets correctly associated with the label.

Reply:
Thanks, fixed that.